5_analyze_data/notebooks/KS-test_analysis/nf1_ks_test.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Perform Two-Sample KS-Test on NF1 Genotype Samples
+
+# ## Import libraries
+
+# In[1]:
+
+
+import numpy as np
+from scipy.stats import ks_2samp
+import pathlib
+import pandas as pd
+
+
+# ## Set seed
+
+# In[2]:
+
+
+np.random.seed(0)
+
+
+# ## Load in NF1 data
+
+# In[3]:
+
+
+norm_fs_data = pathlib.Path("../../../4_processing_features/data/nf1_sc_norm_cellprofiler.csv.gz")
+
+data = pd.read_csv(norm_fs_data, compression="gzip", index_col=0)
+
+
+# ## Helper functions to perform KS-test and create final `csv` file with results
+
+# In[4]:
+
+
+def nf1_ks_test_two_sample(data: pd.DataFrame):
+    """seperate features by genotype and perform two sample ks-test on each feature
+
+    Parameters
+    ----------
+    data : pd.Dataframe
+        pycytominer output after normalization and feature selection
+
+    Returns
+    -------
+    pd.Dataframe
+        feature results from the two sample ks-test
+    """
+    feature_results = []
+
+    null_features = data[(data["Metadata_genotype"] == "Null")]
+    wt_features = data[(data["Metadata_genotype"] == "WT")]
+
+    for column in data:
+        if "Metadata" not in column:
+            null_feature = null_features[column].to_numpy()
+            wt_feature = wt_features[column].to_numpy()
+            results = ks_2samp(wt_feature, null_feature)
+            # have to seperate out namedtuple due to scipy hiding the last two results 
+            results = tuple([results.statistic, results.pvalue, results.statistic_location, results.statistic_sign])
```

Comment:
Looking into this a bit more after our discussions I wanted to follow up with this suggestion. You can access all values here using the `_asdict` method of `result`, see below for more detail.

As a recap: The `ks_2samp` result stores `statistic_location` and `statistic_sign` differently than `statistic` and `pvalue` to help with backwards compatibility (this is a new grouping of fields/values in `scipy==1.10.0`) in a namedtuple-like format referenced as a "tuple bunch" ([found here](https://github.com/scipy/scipy/blob/main/scipy/_lib/_bunch.py#L33)) . This tuple bunch by default will show values from `fields` when accessed as a tuple. It also includes `extra_fields` that require a little more work to access. 

Luckily, the tuple bunch class also includes a method, `_asdict` ([reference](https://github.com/scipy/scipy/blob/main/scipy/_lib/_bunch.py#L172)), which converts all fields as keys and associated values to a dictionary. Using a dict, we may access all values without keys using `.values()`.

```suggestion
            results = tuple(list(result._asdict().values()))
```

Reply:
This worked perfectly! I am glad that this was a learning experience for tuples and getting to learn these methods. Thank you Dave! @d33bs